superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx

```diff
@@ -179,12 +197,14 @@ const DatabaseConnectionForm = ({
   getValidation: () => void;
 }) => (
   <>
-    <StyledFormHeader>
-      <h4>Enter the required {name} credentials</h4>
-      <p className="helper">
-        Need help? Learn more about connecting to {name}.
-      </p>
-    </StyledFormHeader>
+    {!isEditMode && (
+      <StyledFormHeader>
+        <h4>Enter the required {name} credentials</h4>
+        <p className="helper">
+          Need help? Learn more about connecting to {name}.
+        </p>
+      </StyledFormHeader>
+    )}
```

Comment:
you can move the header out to the index file like we did here: https://github.com/apache/superset/blob/e53983404ef797f0da6e6136f0ea01550d1d56f4/superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx#L367